db-connector/src/main/java/uk/nhs/adaptors/connector/service/PatientAttachmentLogService.java

```diff
@@ -0,0 +1,104 @@
+package uk.nhs.adaptors.connector.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.connector.dao.PatientAttachmentLogDao;
+import uk.nhs.adaptors.connector.model.PatientAttachmentLog;
+
+@Slf4j
+@Service
+@AllArgsConstructor(onConstructor = @__(@Autowired))
+
+public class PatientAttachmentLogService {
+    private final PatientAttachmentLogDao patientAttachmentLogDao;
+
+    /**
+     *
+     * @param attachmentLogInput
+     * @description
+     * Requires a minimum of mid, filename and patient request id
+     *
+     */
+    public void addAttachmentLog(
+        PatientAttachmentLog attachmentLogInput
+    ) {
+        var mid = attachmentLogInput.getMid();
+        var filename = attachmentLogInput.getFilename();
+        patientAttachmentLogDao.addAttachmentLog(
+            mid,
+            filename,
+            attachmentLogInput.getParent_mid(),
+            attachmentLogInput.getPatient_migration_req_id(),
+            attachmentLogInput.getContent_type(),
+            attachmentLogInput.getCompressed(),
+            attachmentLogInput.getLarge_attachment(),
+            attachmentLogInput.getBase64(),
+            attachmentLogInput.getSkeleton(),
+            attachmentLogInput.getUploaded(),
+            attachmentLogInput.getOrder_num(),
+            attachmentLogInput.getLength_num()
+        );
+        LOGGER.debug("Created migration log mid=[{}] for patient migration request id=[{}]", mid, filename);
+    }
+
+    /**
+     *
+     * @param mid
+     * @param conversationId
+     * @return PatientAttachmentLog
+     *
+     * @description
+     * Find an attachment log with an mid and conversation id
+     */
+    public PatientAttachmentLog findAttachmentLog(
+        String mid,
+        String conversationId
+    ) {
+        return patientAttachmentLogDao.findPatientAttachment(mid, conversationId);
```

Comment:
should we throw if no log is found? We may also need to make sure that the returned result does not have a deleted flag set to true

Reply:
Do we have any tests for these functions?